code/modules/cm_marines/orbital_cannon.dm

```diff
@@ -419,18 +461,23 @@ var/list/ob_type_fuel_requirements
 	var/datum/cause_data/cause_data = create_cause_data(initial(name), source_mob)
 	cell_explosion(target, clear_power, clear_falloff, EXPLOSION_FALLOFF_SHAPE_LINEAR, null, cause_data) //break shit around
 	sleep(clear_delay)
-	//ACTUALLY BLOW SHIT UP
-	if(!target.density)
+
+	// Explosion if turf is not blocked.
+	if(!is_blocked_turf(target))
```

Comment:
is_blocked_turf is NOT the same as turf.density

It checks for dense atoms on the turf, which means that a lot more things than walls would block OBs now.